slave-setup/worker_installer/slave_setup.sh

```diff
@@ -0,0 +1,265 @@
+#!/bin/sh
+# script that should be executed on the slave after all files are copied
+
+# Installing System Tools
+echo ""
+echo ""
+echo "*****************************************************************************"
+echo "*                            Installing System Tools                        *"
+echo "*****************************************************************************"
+echo ""
+echo ""
+
+# Updating to latest packages
+sudo apt-get update
+
+# Installing Git
+echo ""
+echo ""
+echo "Installing git........................................,"
+echo ""
+echo ""
+sudo apt-get -y install git
+# Installing Apache ANT
+echo ""
+echo ""
+echo "Installing ant........................................,"
+echo ""
+echo ""
+sudo apt-get -y install ant
+# Installing unzip
+echo ""
+echo ""
+echo "Installing zip/unzip..................................,"
+echo ""
+echo ""
+sudo apt-get -y install zip unzip
+# Installing curl
+echo ""
+echo ""
+echo "Installing curl.......................................,"
+echo ""
+echo ""
+sudo apt-get -y install curl
+
+
+
+#Xvfb configuration
+echo ""
+echo ""
+echo "Installing Firefox.......................................,"
+echo ""
+echo ""
+#Install Firefox and downgrade to version 28
+sudo apt-get --yes --force-yes install firefox
+apt-cache show firefox | grep Version
+echo ""
+echo ""
+echo "Downgrade Firefox version to 28.0.......................................,"
+echo ""
+echo ""
+sudo apt-get --yes --force-yes install firefox=28.0+build2-0ubuntu2
+sudo apt-mark hold firefox
+#Install Xvfb X 
+echo "Installing xvfb.......................................,"
+sudo apt-get -y install xvfb
+#Install dbus-x11
+echo "Installing dbus-x11.......................................,"
+sudo apt-get -y install dbus-x11
+
+
+#Creating Directories
+cd /build/jenkins-home
+mkdir -p software/java
+mkdir -p software/maven
+mkdir -p software/jce
+cd
+
+# unzip installation file
+#unzip java files
+echo ""
+echo "*****************************************************************************"
+echo "*                            Extracting Java files                          *"
+echo "*****************************************************************************"
+echo ""
+
+tar -zxvf /build/jenkins-home/slaveSetupFile/jdk-7u51-linux-x64.tar.gz -C /build/jenkins-home/software/java
+tar -zxvf /build/jenkins-home/slaveSetupFile/jdk-8u45-linux-x64.tar.gz -C /build/jenkins-home/software/java
```

Comment:
Have a readme that says user should copy these tar files into $correct folder. 

This need to go in to slave's /build/jenkins-home/slaveSetupFile folder, is it?